modules/mobile/app/controllers/mobile/v0/awards_controller.rb

```diff
@@ -8,6 +8,8 @@ module V0
     class AwardsController < ApplicationController
       def index
         award_data = regular_award_service.get_awards
+        raise Common::Exceptions::BackendServiceException, 'MOBL_502_upstream_error' unless award_data
```

Comment:
FYI, this will also happen if we receive nil from upstream. I assume that's ok, because a normal empty response should be an empty array, not nil, but given that the upstream behavior is already atypical, I'm not sure we can count on that. Do we know if nil is possible? The next line of code would bomb on nil anyhow, so maybe this is for the best regardless. But that leads me to wonder... if we're merging other data into this, does it make sense to raise the error on failure? We could still have a partial success, right? 

Also, do you know what would cause this error? The only reasons I can think of for an index to error would be on things like "user doesn't exist" or "user doesn't have access".

Reply:
yeah, this does have the potential to kill a partial response if we assume the pension awards would succeed where the original call failed. there's no VCRs or anything I can find for any other failure conditions beyond  "user doesn't exist" or "user doesn't have access" so if one fails I'd imagine the other does but just in case, I've made this ticket https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8021 . I think it'll worth changing the schema to group up the two calls into different hashes so it's more obvious which field would be missing in a partial response.